### PR TITLE
Use posix archive format

### PR DESCRIPTION
**Pull request by smorimoto**
Related issue: https://github.com/actions/cache/issues/326

### CODE REVIEW

**Comment by imbsky** [comment]
I've checked that this works on my Windows machine, so there should be no major problem.

**Comment by imbsky** [comment]
@joshmgross @aiqiaoy friendly ping

File: packages/cache/src/internal/tar.ts [hunk 1]
```diff
@@ -62,6 +62,7 @@ export async function extractTar(
     }
   }
   const args = [
+    '--posix',
```
Comment (joshmgross) [inline packages/cache/src/internal/tar.ts:65]: What happens if `--posix` is called on an archive that wasn't created with the `--posix` flag?

Comment (imbsky) [reply packages/cache/src/internal/tar.ts:65]: I will check it.

Comment (OlegYch) [reply packages/cache/src/internal/tar.ts:65]: `--posix` is not allowed when extracting
```
>tar --posix -xzf qwe.tgz qwe                   
tar: Option --format is not permitted in mode -x
```

Comment (imbsky) [reply packages/cache/src/internal/tar.ts:65]: Oops, all right. I will update this PR.

**Comment by imbsky** [comment]
Status?

**Comment by smorimoto** [comment]
Hi @bryanmacfarlane. Can you check this?

**Comment by smorimoto** [comment]
If necessary, please let me know so that I can help you with the maintenance of the cache action related projects.

**Comment by aiqiaoy** [comment]
I'm gonna pull the branch and test on my mac (mac and linux runner is using `zstd` tar)

**Comment by smorimoto** [comment]
Sorry. I will be in an area with an unstable Internet connection until the 29th. Of course, I can work on this later. But, if you don't mind, this branch is set to be editable by you, so feel free to push those changes.

**Comment by smorimoto** [comment]
I lost the repository because of an accident, so I will close this and open another PR later.

**Comment by smorimoto** [comment]
This was merged as https://github.com/actions/toolkit/pull/533.

**Pull request closed without merging**